src/Microsoft.AspNetCore.SignalR.Common/Protocol/HandshakeResponseMessage.cs

```diff
@@ -11,21 +11,41 @@ public class HandshakeResponseMessage : HubMessage
         /// <summary>
         /// An empty response message with no error.
         /// </summary>
-        public static readonly HandshakeResponseMessage Empty = new HandshakeResponseMessage(null);
+        public static readonly HandshakeResponseMessage Empty = new HandshakeResponseMessage(error: null);
 
         /// <summary>
         /// Gets the optional error message.
         /// </summary>
         public string Error { get; }
 
+        /// <summary>
+        /// Highest minor protocol version that the server supports.
+        /// </summary>
+        public int MinorVersion { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HandshakeResponseMessage"/> class.
+        /// An error response does need a minor version. Since the handshake has failed, any extra data will be ignored.
+        /// </summary>
+        /// <param name="error"></param>
+        public HandshakeResponseMessage(string error) : this(null, error) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HandshakeResponseMessage"/> class.
+        /// A reponse with a minor version indicates success, and doesn't require an error field.
+        /// </summary>
+        /// <param name="minorVersion"></param>
+        public HandshakeResponseMessage(int minorVersion) : this(minorVersion, null) { }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="HandshakeResponseMessage"/> class.
         /// </summary>
-        /// <param name="error">An optional response error message. A <c>null</c> error message indicates a succesful handshake.</param>
-        public HandshakeResponseMessage(string error)
+        public HandshakeResponseMessage(int? minorVersion, string error)
```

Comment:
I'd rather this took an `int` and that the caller of the constructor does the `?? 0` logic. As a general rule I think it's good when constructor parameters map directly to the type and name of the property, unless there's a really good reason to do that. It means the caller has to do some more work, but that's actually a good thing here I think because it lets the caller decide what a "missing" minor version represents.

Reply:
Ping on this comment :)